src/clusterd/src/lib.rs

```diff
@@ -141,15 +141,29 @@ struct Args {
     worker_core_affinity: bool,
 }
 
-#[tokio::main]
-pub async fn main() {
+pub fn main() {
     mz_ore::panic::install_enhanced_handler();
 
     let args = cli::parse_args(CliConfig {
         env_prefix: Some("CLUSTERD_"),
         enable_version_flag: true,
     });
-    if let Err(err) = run(args).await {
+
+    let ncpus_useful = usize::max(1, std::cmp::min(num_cpus::get(), num_cpus::get_physical()));
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .worker_threads(ncpus_useful)
+        // The default thread name exceeds the Linux limit on thread name
+        // length, so pick something shorter.
```

Comment:
Could you mention the limit in this comment too? Might be useful in case someone wants to touch this code in the future.